### ISSUE TITLE
Mail configuration confusing for docker installation

### ISSUE THREAD

**Issue by daliborfilus** [getsentry/self-hosted#1960]
### Self-Hosted Version

23.1.1

### CPU Architecture

x86_64

### Docker Version

23.0.1

### Docker Compose Version

1.29.2

### Steps to Reproduce

* Download get-sentry installation.
* Edit .env to add `SENTRY_MAIL_HOST=mail.our-domain.com` (obfuscated for this issue, I will use it consistently), because you think that's the place where you should configure sentry's outgoing mail.
* Boot Sentry up.
* Verify mail configuration in https://sentry.our-domain.com/manage/status/mail/ is valid. Discover you want Sentry to use proper login, so let's edit the config directly:
* Edit `sentry/config.yml` and add our mail configuration:

```
mail.host: 'mail.our-domain'
mail.port: 25
mail.username: 'sentry@our-domain.com'
mail.password: 'pass'
mail.from: 'sentry@our-domain.com'
mail.use-tls: true
```

* Restart sentry using `docker-compose restart web`.
* Go to admin panel, verify this is correct (it is):

```
From Address

    sentry@our-domain.com

Host

    mail.our-domain.com:25

Username

    sentry@our-domain.com

Password

    ********

STARTTLS?

    Yes

SSL?

    No
```

* Try to send email.
* Get **STARTTLS extension not supported** error message. Be confused, because you OWN the mail server, you verified it using telnet & STARTTLS. Verify it using `openssl s_client -startls smtp ...`, etc., you KNOW it works on that port.
* Get distracted by this nonsensical, "impossible" error. Try another combinations, try other combinations of ports and SSL instead of TLS, etc., all other ports (465, 587) now return "Error 111: connection refused". But why? The ports are OPEN and are working, as verified from the docker host. Test some more just to be sure, to no avail.
* Try to configure sentry debug logging in `.env`, somehow `SENTRY_LOG_LEVEL` doesn't have any effect. The log is very short for the test mail button.

* Finally arrive at the conclusion it must be something within the container.
* Go in, `docker-compose exec web bash`, `apt-get update && apt-get install iputils-ping`. Issue ping to `mail.our-domain.com` and discover it pings another container, instead of your own mail server.
* Open `docker-compose.yml` and discover that `SMTP_MAIL_HOST` is used as a hostname for `smtp` container. That means all your pings and test mails were sent to this other container, even though the UI is not aware of that and shows `mail.our-domain.com` as the SMTP server.
* Go to github and search for when this started.
* Find commit https://github.com/getsentry/self-hosted/commit/b32de84d31dd1cc0b9c9534fae2e95ad86a0cbee two yerars ago.
* Remove `smtp:` service from docker-compose.yml and it finally works. Wow.

TL;DR: `SENTRY_MAIL_HOST` is used for something completely different than I thought and is confusing. A little more documentation on that would be helpful.

In ideal scenario, there should be some way for using your own existing mail server like I want and removing the `smtp` container entirely using this configuration. Or at least some warning about what this env. variable really does.

I understand that I broke it by the combination of using `SENTRY_MAIL_HOST` AND editing config file directly, but the situation led to it naturally, because I wanted to customize the port and it wasn't (visibly) possible in `.env` file, I had to resort to editing config file directly, but left my edited `SENTRY_MAIL_HOST` in place.

If you google for "sentry starttls extension not supported" there are some results and some may come from the same confusion.

### Expected Result

Outgoing mail is working and setting up outgoing mail server should not be confusing.

### Actual Result

See above

**Comment by linpan** [comment]
metoo. sentry is kill us

**Comment by hubertdeng123** [comment]
Got it, sorry you had so much trouble setting this up. I can add information in our [docs](https://develop.sentry.dev/self-hosted/email/) and link it in the config.yml. 

I see this but it can be unclear, so I'll add some more documentation around this
```
If you want to use an external SMTP server you can set the relevant mail.* settings in config.yml file and ignore the built-in SMTP server.
```

Thanks for bringing this up

**Comment by linpan** [comment]
@hubertdeng123 
Error response from daemon: driver failed programming external connectivity on endpoint sentry-self-hosted-nginx-1 (f8144034b55ca3f44a087ba5e1eebd5331b5ce3f0b1d32a8c0d5f695f79d21f6): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use
[linpan@localhost self-hosted]$ sudo netstat -ntlp |grep 9000
tcp        0      0 127.0.0.1:9000          0.0.0.0:*               LISTEN      51737/clickhouse-se
tcp6       0      0 ::1:9000                :::*                    LISTEN      51737/clickhouse-se
